Fixed several issues with cropping.

jeffqjiangNew opened the pull request:
- Fixed the pixel start offset error on streams with non-zero top/left cropping in MD5 calculation.
  - Fixed the pixel start offset error on streams with non-zero top/left cropping, and/or with user specified cropping in YUV frame output.
  - Sorted out code which deals with cropping area specifiied by the bitstream, and cropping area specified by the user.

jeffqjiangNew committed:
* rocDecode: Fixed several issues with cropping.
  - Fixed the pixel start offset error on streams with non-zero top/left cropping in MD5 calculation.
  - Fixed the pixel start offset error on streams with non-zero top/left cropping, and/or with user specified cropping in YUV frame output.
  - Sorted out code which deals with cropping area specifiied by the bitstream, and cropping area specified by the user.

AryanSalmanpour committed:
Merge branch 'develop' into jj/cropping_fix

rrawther reviewed (commented):
I think we need to discuss this before merging. Let us have a call sometime

jeffqjiangNew committed:
Merge branch 'develop' into jj/cropping_fix

jeffqjiangNew committed:
Merge branch 'ROCm:develop' into jj/cropping_fix

jeffqjiangNew committed:
Merge branch 'jj/cropping_fix' of github.com:jeffqjiangNew/rocDecode into jj/cropping_fix

rrawther reviewed (commented):
I think we shouldn't modify target rectangle with crop dimentions.
@jeffqjiangNew Let me know if you want to have a discussion

jeffqjiangNew committed:
* rocDecode: Additional changes after discussion.
 - Set display rect for decoder create/reconfig to displayable area or user specified ROI area.
 - Do not set target rect for decoder create/reconfig. Leave it for future use.
 - Unified naming of display rect in decoder create/reconfig structs.

rrawther reviewed (commented):
Added few minor comments

jeffqjiangNew committed:
* rocDecode: Put back the extra rounding to target width/height specifiied by the coded stream, for peace of mind, based on code review.

jeffqjiangNew committed:
* rocDecode: Added the missing stream specified cropping offset to the starting pixel pointer for none-internal memory modes.

jeffqjiangNew committed:
Merge branch 'develop' into jj/cropping_fix

jeffqjiangNew committed:
Merge branch 'develop' into jj/cropping_fix
